devito/ir/ietxdsl/cluster_to_ssa.py

```diff
@@ -559,8 +568,16 @@ def convert(self, eqs: Iterable[Eq], **kwargs) -> builtin.ModuleOp:
             functions = OrderedSet()
             for eq in eqs:
                 if isinstance(eq, Eq):
-                    for f in retrieve_functions(eq):
+                    funcs = retrieve_function_carriers(eq)
+                    carriers = retrieve_function_carriers(eq)
+                    try:
+                        assert funcs == carriers
```

Comment:
Is it supposed to use a different retriever? You're using the same function twice here